### ISSUE TITLE
NVDA crashes when reading some non-latin characters letter by letter

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4553]
**Reported by Sukil on 2014-10-15 13:25**
STR:
Make sure the speech synthesizer in use is eSpeak, and that it switches languages automatically.
1. Go to the article on "Great expectations"in wikipedia.
2. Go to the languages heading, and down arrow. I'm not sure till when, but I know I pased something like "u u".
3. By using down arrow, fin a language witch is non-latin, and in which eSpeak changes language.
4. Right arrow rapidly.

**Comment by rhdunn** [comment]
Hi, I have tried that SSML string on my espeak-ng linux build and espeak 1.48.15 and get:

    (en)b'a55ksla55S_!(zh)_| j'iou55_!_| s'i[51_!_| j'i51_!_| '@r51_!_| t'i51_!_|

Looking at the code, I can't see anywhere it is handling `\uNNNN` style character markup. Additionally,  the SSML documents for the say-as attribute values don't mention that syntax.

__NOTE:__ Using `&#x4e2d;` results in espeak saying (with the default, incomplete dictionary):

    ts.'ong55_!_|

What compiler and settings are you using on Windows?

**Comment by jcsteh** [comment]
Thanks for looking at this @rhdunn.

> Looking at the code, I can't see anywhere it is handling `\uNNNN` style character markup. Additionally,  the SSML documents for the say-as attribute values don't mention that syntax.

Oops. I must have copied the raw string from Python. That should be:

```
<voice xml:lang="zh"><say-as interpret-as="characters">中</say-as></voice>
```

> __NOTE:__ Using `&#x4e2d;` results in espeak saying (with the default, incomplete dictionary):
> 
>     ts.'ong55_!_|
> 
> What compiler and settings are you using on Windows?

@MichaelDCurran?

**Comment by michaelDCurran** [comment]
The crash no longer exists in eSpeak NG master, probably through some msc /analyze fixes. Therefore this will be fixed once #5651 is merged.

**Comment by jcsteh** [comment]
#5651 is now merged. Closing.
